packages/mobile/src/exchange/ExchangeHomeScreen.tsx

```diff
@@ -1,126 +1,231 @@
+import { goldToDollarAmount } from 'src/utils/currencyExchange'
+import { getLocalCurrencyDisplayValue } from 'src/utils/formatting'
 
-interface StateProps {
-  exchangeRate: BigNumber
-  goldBalance: string | null
-  dollarBalance: string | null
-}
+type Props = StackScreenProps<StackParamList, Screens.ExchangeHomeScreen>
 
-interface DispatchProps {
-  fetchExchangeRate: typeof fetchExchangeRate
-}
+function ExchangeHomeScreen({ navigation }: Props) {
+  const scrollPosition = useRef(new Animated.Value(0)).current
+  const { t } = useTranslation(Namespaces.exchangeFlow9)
+  const dollarBalance = useSelector((state) => state.stableToken.balance)
+  const goldBalance = useSelector((state) => state.goldToken.balance)
+  const dispatch = useDispatch()
 
-type OwnProps = StackScreenProps<StackParamList, Screens.ExchangeHomeScreen>
+  // TODO: revert this back to `useLocalCurrencyCode()` when we have history data for cGDL to Local Currency.
+  const localCurrencyCode = null
+  const localExchangeRate = useSelector(getLocalCurrencyExchangeRate)
+  const currentExchangeRate = useExchangeRate()
+  const goldToDollars = useCallback((amount) => goldToDollarAmount(amount, currentExchangeRate), [
+    currentExchangeRate,
+  ])
+  const dollarsToLocal = useCallback(
+    (amount) => convertDollarsToLocalAmount(amount, localCurrencyCode ? localExchangeRate : 1),
+    [localExchangeRate]
+  )
+  const displayLocalCurrency = useCallback(
+    (amount: BigNumber.Value) =>
+      getLocalCurrencyDisplayValue(amount, localCurrencyCode || LocalCurrencyCode.USD, true),
+    [localCurrencyCode]
+  )
 
-type Props = StateProps & DispatchProps & WithTranslation & OwnProps
+  const currentGoldRateInLocalCurrency = dollarsToLocal(goldToDollars(1))
+  let rateChangeInPercentage, rateWentUp
+  const exchangeHistory = useSelector(exchangeHistorySelector)
+  if (exchangeHistory.aggregatedExchangeRates.length) {
+    const oldestGoldRateInLocalCurrency = dollarsToLocal(
+      exchangeHistory.aggregatedExchangeRates[0].exchangeRate
+    )
+    if (oldestGoldRateInLocalCurrency) {
+      const rateChange = currentGoldRateInLocalCurrency?.minus(oldestGoldRateInLocalCurrency)
+      rateChangeInPercentage = currentGoldRateInLocalCurrency
+        ?.div(oldestGoldRateInLocalCurrency)
+        .minus(1)
+        .multipliedBy(100)
+      rateWentUp = rateChange?.gt(0)
+    }
+  }
 
-const mapStateToProps = (state: RootState): StateProps => ({
-  exchangeRate: getRateForMakerToken(state.exchange.exchangeRatePair, CURRENCY_ENUM.DOLLAR),
-  goldBalance: state.goldToken.balance,
-  dollarBalance: state.stableToken.balance,
-})
+  const hasGold = new BigNumber(goldBalance || 0).isGreaterThan(0)
 
-export class ExchangeHomeScreen extends React.Component<Props> {
-  componentDidMount() {
-    this.props.fetchExchangeRate()
-  }
+  useEffect(() => {
+    dispatch(fetchExchangeRate())
+  }, [])
+
+  const headerOpacity = useMemo(
```

Comment:
i'm a little confused about the necessity of useMemo here

Reply:
As @jeanregisser pointed out:
> When using Animated.View, it's more performant to pass a style that doesn't change.

My understanding, that we treat `Animated.View` as an equivalent to `React.memo(...)`,
so, that's why we are memoing the style here.

Reply:
Yes exactly. It has to recreate animation nodes if they are different on each render.
You can see it by following what it does here https://github.com/software-mansion/react-native-reanimated/blob/1.9.0/src/createAnimatedComponent.js#L141-L161
Also more discussion here: https://github.com/software-mansion/react-native-reanimated/issues/267